PhysicsExploration/GravitySimulation.cpp

```diff
@@ -0,0 +1,64 @@
+#include "GravitySimulation.h"
+
+float GravitySimulation::u;
+float GravitySimulation::v;
+float GravitySimulation::s0;
+float GravitySimulation::s;
+Gravity* GravitySimulation::gravity;
+void GravitySimulation::initializeShaders() const
+{
+	_shader = new Shader("../PhysicsExploration/planet_vertex.shader", "../PhysicsExploration/planet_fragment.shader");
+	_lightShader = new Shader("../PhysicsExploration/light_vertex.shader", "../PhysicsExploration/light_fragment.shader");
+	_depthShader = new Shader("../PhysicsExploration/shadow_vertex.shader", "../PhysicsExploration/shadow_fragment.shader");
+}
+
+GravitySimulation::GravitySimulation()
+{
+	u = 0.0f; 
+	v = 0.0f; 
+	s0 = 0.0f; 
+	s = 0.0f;   
+	gravity = new Gravity();
+}
+
+void GravitySimulation::initializeModels() const
+{
+	_plane = new Model("../External Resources/Models/plane.obj");
+	_object = new Model("../External Resources/Models/planet_obj.obj");
+}
+
+GravitySimulation::~GravitySimulation()
+{
+}
+
+void GravitySimulation::renderScene(const Shader& shader) const
+{
+	shader.setVector("spotLight.position", glm::vec4(_camera->GetPosition(), 1.0f));
+	shader.setVector("spotLight.direction", _camera->GetFront());
+	shader.setVector("view_position", _camera->GetPosition());
+	updatePhysicsParameters();
```

Comment:
I don't see the new position s0 being applied anywhere? am I missing something?

Reply:
Actually s is being used if you check code properly instead s0. s0 is their to keep track of last s.

Reply:
Actually, declaring operands while in frame is taking too much of time to allocate memory. I remember why I declare them all global Rose :)

Reply:
interesting!

Reply:
but that still didn't answer my question. Where is the new position s (or s0) being applied?

Reply:
> but that still didn't answer my question. Where is the new position s (or s0) being applied?

Would you like to search following line in a file ? see it uses s0. 
model_matrix = glm::translate(model_matrix, glm::vec3(0.0f, -1.0f * s0, 0.0f));

Reply:
I didn't see it, thanks! :P